### ISSUE TITLE
Conditional textures for nodes, bulk texture changing

### ISSUE THREAD

**Issue by kromka-chleba** [minetest/minetest#13417]
## Problem

TL;DR I want to change ALL grass textures from green spring to dark winter, but ABM/LBMs are slow and bad at the task

I added seasonal changes for Exile (https://content.minetest.net/packages/Mantar/exile/), but the ABM/LBM implementation we used is heavy on resources and inefficient as described in #13112. An alternative solution came to my mind. See below.

## Solutions

My problem would be solved if I could swap node textures instead of swapping nodes during runtime.
Would it be possible to implement this? If implemented, would performance be less affected compared to running ABMs on each soil node?
Something like this:

```
if seasons.is_winter() then
    minetest.swap_node_texture("nodes_nature:barrenland_soil", "nodes_nature_barrenland_soil_winter.png")
end

-- or

if seasons.is_winter() then
    minetest.swap_node_texture("nodes_nature:barrenland_soil", 2)
    -- 2 is the number of an alternative texture
end
```
After running that code, the game would "instantly" swap the texture for all loaded nodes.

This would make changing appearance of the landscape easier.

## Alternatives

An alternative solution is here #13112

**Comment by Desour** [comment]
It would of course be *possible*, but it's non-trivial. The client would have to track where the node tiles are copied to, and update block meshes and whatnot. Nodedef would no longer be strictly read-only after init.

**Comment by wsor4035** [comment]
tangentially related: https://github.com/minetest/minetest/pull/12928

**Comment by kromka-chleba** [comment]
Also it would be good if minetest.swap_node_texture worked only on nodes with that option specified in their node definitions. This way we would avoid unnecessary overhead for normal nodes.

**Comment by x2048** [comment]
Until we have texture atlas and provided all textures are the same dimensions this could be done by mapping tile material to a render target texture instead of fixed image texture, and then drawing images into that render target, depending on the selection of the texture. Can work instantaneously, unlike remeshing.

**Comment by Desour** [comment]
No core-dev has explicitly expressed support for this. Marked as low priority. Don't expect large PRs for this to be easily merged.

https://irc.minetest.net/minetest-dev/2023-04-30#i_6080402
